packages/data-context/test/unit/actions/ProjectActions.spec.ts

```diff
@@ -17,6 +18,51 @@ describe('ProjectActions', () => {
     actions = new ProjectActions(ctx)
   })
 
+  context('getReactComponentsFromFile', () => {
+    const absolutePathPrefix = path.resolve('./test/unit/actions/project')
+
+    it('returns React components from file with class component', async () => {
+      const reactComponents = await actions.getReactComponentsFromFile(`${absolutePathPrefix}/counter-class.jsx`)
+
+      expect(reactComponents).to.have.length(1)
+      expect(reactComponents[0].displayName).to.equal('Counter')
+    })
+
+    it('returns React components from file with functional component', async () => {
+      const reactComponents = await actions.getReactComponentsFromFile(`${absolutePathPrefix}/counter-functional.jsx`)
+
+      expect(reactComponents).to.have.length(1)
+      expect(reactComponents[0].displayName).to.equal('Counter')
+    })
+
+    it('returns only exported React components from file with functional components', async () => {
+      const reactComponents = await actions.getReactComponentsFromFile(`${absolutePathPrefix}/counter-multiple-components.jsx`)
+
+      expect(reactComponents).to.have.length(1)
+      expect(reactComponents[0].displayName).to.equal('CounterContainer')
+    })
+
+    it('returns React components from a tsx file', async () => {
+      const reactComponents = await actions.getReactComponentsFromFile(`${absolutePathPrefix}/counter.tsx`)
+
+      expect(reactComponents).to.have.length(1)
+      expect(reactComponents[0].displayName).to.equal('Counter')
+    })
+
+    it('returns React components that are exported by default', async () => {
+      const reactComponents = await actions.getReactComponentsFromFile(`${absolutePathPrefix}/counter-default.tsx`)
+
+      expect(reactComponents).to.have.length(1)
+      expect(reactComponents[0].displayName).to.equal('CounterDefault')
+    })
+
+    it('does not throw while parsing empty file', async () => {
+      const reactComponents = await actions.getReactComponentsFromFile(`${absolutePathPrefix}/empty.jsx`)
+
+      expect(reactComponents).to.have.length(0)
+    })
```

Comment:
As we discussed @astone123 , there are a few conditions that aren't covered here that I would like to see tests added for:

1. Multiple named exports are parsed
2. Mix of named and default exports are parsed
3. Functional component that returns a non-JSX value is parsed (namely, a string) - appears to be a limitation of the docgen library
4. Functional component that returns null is parsed - appears to be a limitation of the docgen library

Reply:
I'd like to try to handle 3 and 4 in https://github.com/cypress-io/cypress/issues/24929 since I think we'll need to use a custom resolver for both of these issues. As of right now I'm considering 3 and 4 as edge cases.

Will add test cases for 1 and 2